skyportal/handlers/api/photometry.py

```diff
@@ -166,489 +172,548 @@ def serialize(phot, outsys, format):
                 )
-                .scalars()
-                .first()
-            )
-            group_ids = [public_group.id]
-        else:
-            raise ValidationError(
-                "Invalid group_ids parameter value. Must be a list of IDs "
-                "(integers) or the string 'all'."
+            group = Group.query.get(group_id)
+            if group is None:
+                raise ValidationError(f'No group with ID {group_id}')
+    elif group_ids == 'all':
+        public_group = (
+            DBSession()
+            .execute(
+                sa.select(Group).filter(Group.name == cfg["misc"]["public_group_name"])
             )
+            .scalars()
+            .first()
+        )
+        group_ids = [public_group.id]
+    else:
+        raise ValidationError(
+            "Invalid group_ids parameter value. Must be a list of IDs "
+            "(integers) or the string 'all'."
+        )
 
-        # always add the single user group
-        group_ids.append(self.associated_user_object.single_user_group.id)
-        group_ids = list(set(group_ids))
-        return group_ids
+    # always add the single user group
+    group_ids.append(user.single_user_group.id)
+    group_ids = list(set(group_ids))
+    return group_ids
 
-    def get_stream_ids(self):
-        data = self.get_json()
-        stream_ids = data.pop("stream_ids", [])
-        if isinstance(stream_ids, (list, tuple)):
-            for stream_id in stream_ids:
-                try:
-                    stream_id = int(stream_id)
-                except TypeError:
-                    raise ValidationError(
-                        f"Invalid format for stream id {stream_id}, must be an integer."
-                    )
-                stream = Stream.get_if_accessible_by(stream_id, self.current_user)
-                if stream is None:
-                    raise ValidationError(f'No stream with ID {stream_id}')
-        else:
-            raise ValidationError(
-                "Invalid stream_ids parameter value. Must be a list of IDs (integers)."
+
+def get_stream_ids(data, user):
+    stream_ids = data.pop("stream_ids", [])
+    if isinstance(stream_ids, (list, tuple)):
+        for stream_id in stream_ids:
+            try:
+                stream_id = int(stream_id)
+            except TypeError:
+                raise ValidationError(
+                    f"Invalid format for stream id {stream_id}, must be an integer."
+                )
+            stream = Stream.get_if_accessible_by(stream_id, user)
+            if stream is None:
+                raise ValidationError(f'No stream with ID {stream_id}')
+    else:
+        raise ValidationError(
+            "Invalid stream_ids parameter value. Must be a list of IDs (integers)."
+        )
+
+    stream_ids = list(set(stream_ids))
+    return stream_ids
+
+
+def add_external_photometry(json, user):
+    """
+    ---
```

Comment:
This docstring seems to be misplaced

Reply:
what does that mean?

Reply:
Ah I see, because now we just return the log()

Reply:
Well, it looks like a handler (`GET`, `POST`, `PUT`, etc.) docstring in YAML, whereas I would expect a standard Python function docstring that describes the two arguments: `json` and `user`.

Reply:
Ah yes, hopefully more clear now?